bin/templates/scripts/cordova/lib/run.js

```diff
@@ -0,0 +1,184 @@
+        return build.run(argv);
+    }).then(function () {
+
+        var xcodeProjFiles = shell.ls(projectPath).filter(function (name) {
+            return path.extname(name) === '.xcodeproj';
+        });
+        
+        if (xcodeProjFiles.length === 0) {
+            return Q.reject("No Xcode project found in " + projectPath);
+        }
+        if (xcodeProjFiles.length > 1) {
+            console.warn('Found multiple .xcodeproj directories in \n' +
+                projectPath + '\nUsing first one');
+        }
+
+        var projectName = path.basename(xcodeProjFiles[0], '.xcodeproj');
+
+        var appPath = path.join(projectPath, 'build', (args.device ? 'device' : 'emulator'), projectName + '.app');
+        // select command to run and arguments depending whether
+        // we're running on device/emulator
+        if (args.device) {
+            return checkDeviceConnected().then(function () {
+                return deployToDevice(appPath);
+            }).fail(function () {
+                // if device connection check failed use emulator then
+                return deployToSim(appPath, args.target);
+            });
+        } else {
+            return deployToSim(appPath, args.target);
+        }
+    });
+};
+
+/**
+ * Checks if any iOS device is connected
+ * @return {Promise} Fullfilled when any device is connected, rejected otherwise
+ */
+function checkDeviceConnected() {
+    return spawn('ios-deploy', ['-c']);
+}
+
+/**
+ * Deploy specified app package to connected device
+ * using ios-deploy command
+ * @param  {String} appPath Path to application package
+ * @return {Promise}        Resolves when deploy succeeds otherwise rejects
+ */
+function deployToDevice(appPath) {
+    // Deploying to device...
+    return spawn('ios-deploy', ['-d', '-b', appPath]);
+}
+
+/**
+ * Deploy specified app package to ios-sim simulator
+ * @param  {String} appPath Path to application package
+ * @param  {String} target  Target device type
+ * @return {Promise}        Resolves when deploy succeeds otherwise rejects
+ */
+function deployToSim(appPath, target) {
+    // Select target device for emulator. Default is 'iPhone-6' 
+    if (!target) {
+        target = 'iPhone-6';
+        console.log('No target specified for emulator. Deploying to ' + target + ' simulator');
+    }
+    var logPath = path.join(cordovaPath, 'console.log');
+    var simArgs = ['launch', appPath,
+        '--devicetypeid', 'com.apple.CoreSimulator.SimDeviceType.' + target,
+        // We need to redirect simulator output here to use cordova/log command
+        // TODO: Is there any other way to get emulator's output to use in log command?
+        '--stderr', logPath, '--stdout', logPath,
+        '--exit'];
+    return spawn('ios-sim', simArgs);
+}
+
+function listDevices() {
+    var listDevicesScript = path.join(cordovaPath, 'lib', 'list-devices');
+    console.log("Available iOS Devices:");
+    return spawn(listDevicesScript);
```

Comment:
This because these scripts hadn't been ported. Converted them to Node in https://github.com/MSOpenTech/cordova-ios/commit/e4c200b79a9bf5903c07eb70d50841ed328a71cc